File: Problem Set 1/mario-less.c
/**
 * A program that prints out a half-pyramid 
 * of a specified height 
 * (less difficult)
 * @author Ermakov Stepan
*/

#include "cs50.h"
#include <stdio.h>

int main(void)
{
    // Height from 1 to 8
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    for (int row = 1; row <= height; row++)
    {
        for (int number_of_spaces = 0; number_of_spaces < (height - row); number_of_spaces++)
        {
            printf(" ");
        }

        for (int number_of_hashes = 0; number_of_hashes < row; number_of_hashes++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: Problem Set 6/mario-less.py
# A program that prints out a half-pyramid
# of a specified height
# (less difficult)
__author__ = "Ermakov Stepan"


def main():
    height = get_int()
    for x in range(height):
        for y in range(height - (x + 1)):
            print(end=" ")
        print("#" * (x + 1))


def get_int():
    while True:
        try:
            n = int(input("Height: "))
            if (n > 0 and n < 9):
                break
        except ValueError:
            print("That's not an integer!")
    return n


if __name__ == "__main__":
    main()